src/AppInstallerCLICore/TableOutput.h

```diff
@@ -189,11 +189,19 @@ namespace AppInstaller::CLI::Execution
 
                 if (col.MaxLength)
                 {
-                    size_t valueLength = Utility::UTF8Length(line[i]);
+                    size_t valueLength = Utility::UTF8TerminalLength(line[i]);
 
                     if (valueLength > col.MaxLength)
                     {
-                        out << Utility::UTF8Substring(line[i], 0, col.MaxLength - 1);
+                        std::string_view trimmedOutput = Utility::UTF8TrimRightToTerminalLength(line[i], col.MaxLength - 1);
+                        out << trimmedOutput;
+
+                        // Since some characters take 2 unit space, the trimmed string length might be 1 less than the expected length.
+                        if (Utility::UTF8TerminalLength(trimmedOutput) != col.MaxLength - 1)
+                        {
+                            out << ' ';
+                        }

```

Comment:
This should happen after the ellipsis, and it would be great if UTF8TrimRightToTerminalLength could tell us the actual trimmed column width rather than recalculating it (out param?).